docs/knowledge_base/tutorials/chatbot/index.md

```diff
@@ -0,0 +1,288 @@
+- `current_user_message` is the current message that the user is typing
+
+## Step 5: Create a function to generate responses
+
+**This step is the one that needs to be adapted if you want to 
+use a different LLM.**
+
+Initialize the OpenAI client with your API key. You can find 
+your API key [here](https://platform.openai.com/api-keys).
+
+```python
+client = openai.Client(api_key="ENTER_YOUR_API_KEY_HERE")
+```
+
+Create a function that takes as input a string `prompt` which 
+is the user message and returns a string which is the response from the LLM.
+
+```python
+def request(state: State, prompt: str) -> str:
+    """
+    Send a prompt to the GPT-3 API and return the response.
+
+    Args:
+        - state: The current state of the app.
+        - prompt: The prompt to send to the API.
+
+    Returns:
+        The response from the API.
+    """
+    response = state.client.chat.completions.create(
+        messages=[
+            {
+                "role": "user",
+                "content": f"{prompt}",
+            }
+        ],
+        model="gpt-3.5-turbo",
+    )
+    return response.choices[0].message.content
+```
+
+## Step 6: Create a function to add the new messages to the conversation
+
+Create a function that gets triggered when the user sends a 
+message. This function will add the user's message to the context, 
+send it to the API, get the response, add the response to the 
+context and to the displayed conversation.
+
+```python
+def send_message(state: State) -> None:
+    """
+    Send the user's message to the API and update the conversation.
+
+    Args:
+        - state: The current state of the app.
+    """
+    # Add the user's message to the context
+    state.context += f"Human: \n {state.current_user_message}\n\n AI:"
+    # Send the user's message to the API and get the response
+    answer = request(state, state.context).replace("\n", "")
+    # Add the response to the context for future messages
+    state.context += answer
+    # Update the conversation
+    conv = state.conversation._dict.copy()
+    conv["Conversation"] += [state.current_user_message, answer]
+    state.conversation = conv
+    # Clear the input field
+    state.current_user_message = ""
+```
+
+## Step 7: Create the User Interface
+
+In Taipy, one way to define pages is to use Markdown strings. Here we use a 
+[table](../../../manuals/gui/viselements/table.md_template) to display the
+ `conversation` dictionary and an 
+ [input](../../../manuals/gui/viselements/input) so that the 
+ user can type their message. When the user presses enter, 
+ the `send_message` function is triggered.
+
+```python
+page = """
+<|{conversation}|table|show_all|width=100%|>
+<|{current_user_message}|input|label=Write your message here...|on_action=send_message|class_name=fullwidth|>
+"""
+```
+
+## Step 8: Run the application
+
+Finally we run the application:
+
+```python
+Gui(page).run(dark_mode=True, title="Taipy Chat")
```

Comment:
Please specify that this code should be in a if block.
`if __name__ == "__main__":`

Reply:
Fixed